etc/kayobe/ansible/prometheus-network-names.yml

```diff
@@ -0,0 +1,40 @@
+- name: Prometheus friendly network names
+  hosts: overcloud
+  gather_facts: no
+  tasks:
+    - name: Gather network maps from each host with unique identifiers
+      set_fact:
+        host_network_maps: >-
+          {%- set if_list = [] -%}
+          {%- for i in network_interfaces -%}
+            {%- set device_name = hostvars[inventory_hostname][i ~ '_interface'] -%}
+            {%- set friendly_name = i -%}
+            {%- set unique_id = device_name ~ '_' ~ friendly_name -%}
+            {%- set _ = if_list.append({
+              'unique_id': unique_id,
+              'device_name': device_name,
+              'friendly_name': friendly_name
+            }) -%}
+          {%- endfor -%}
+          {{ if_list }}
+      delegate_to: "{{ inventory_hostname }}"
+
+    - name: Aggregate network maps from all hosts
+      set_fact:
+        prometheus_network_maps_aggregated: "{{ groups['overcloud'] | map('extract', hostvars, 'host_network_maps') | flatten }}"
+      run_once: true
+      delegate_to: localhost
+
+    - name: Deduplicate the aggregated list based on unique IDs
+      set_fact:
+        prometheus_network_maps_blob: "{{ prometheus_network_maps_aggregated | unique(attribute='unique_id') }}"
```

Comment:
The issue with this approach is that the map is global, but different hosts may have different uses for interfaces of the same name. It'll probably work fine in most cases, but could lead to confusing results if there's a conflict. The need to deduplicate highlights the issue.

Reply:
Agreed. The only solution I saw to this issue would be to have a seperate scrape_config job for each distinct host network config.
This would also resolve the need to deduplicate.
I had thought this could be implemented with support for discrete files per scrape_config, which would also reduce the maintance burden for the templates.
This is now supported in prometheus upstream https://github.com/prometheus/prometheus/pull/12019 but would be outside the scope of this MR.

Reply:
The other approach considered was to have a pull through wrapper around node-exporter, that could augment specific metrics with additional labels based on some static, host specific config. All other metrics would just pull though unaffected.

That would solve the issue with host specific network interface naming, and re-labelling complexity, but it's a larger task.  It could follow on from INFRA-388.

There was also the approach of just naming network interfaces with a human readable name, rather than the VLAN, but it was considered too invasive (although supported on Ubuntu as I understand).